### ISSUE TITLE
Schema differences after migaration (migration docs bug?) - indexes, not null

### ISSUE THREAD

**Issue by Eccenux** [quartz-scheduler/quartz#932]
The migration guide was added in v2.3:
https://github.com/quartz-scheduler/quartz/blob/quartz-2.3.x/docs/migration-guide.adoc

However, the SQL for creating a database has a different set of indexes:
https://github.com/quartz-scheduler/quartz/blob/quartz-2.3.x/quartz-core/src/main/resources/org/quartz/impl/jdbcjobstore/tables_postgres.sql

To get from `migration-guide.adoc` to SQL in `jdbcjobstore/tables_postgres.sql` you need to run this:
```
-- qrtz_job_details
DROP INDEX IF EXISTS idx_qrtz_j_req_recovery;

-- qrtz_triggers
DROP INDEX IF EXISTS idx_qrtz_t_next_fire_time;
DROP INDEX IF EXISTS idx_qrtz_t_state;
DROP INDEX IF EXISTS idx_qrtz_t_nft_st;

-- qrtz_fired_triggers
DROP INDEX IF EXISTS idx_qrtz_ft_trig_inst_name;
DROP INDEX IF EXISTS idx_qrtz_ft_job_group;
DROP INDEX IF EXISTS idx_qrtz_ft_job_name;
DROP INDEX IF EXISTS idx_qrtz_ft_job_req_recovery;
DROP INDEX IF EXISTS idx_qrtz_ft_trig_group;
DROP INDEX IF EXISTS idx_qrtz_ft_trig_name;
DROP INDEX IF EXISTS idx_qrtz_ft_trig_nm_gp;
```
So basically there are probably too many indexes in [the migration guide](https://github.com/quartz-scheduler/quartz/blob/quartz-2.3.x/docs/migration-guide.adoc). Not sure if that is true for most usages, but for our tenants, qrtz_fired_triggers is empty, so actually no indexes are needed. Your mileage may vary (YMMV).

Also, these columns were added without the NOT NULL constraint in the migration guide:
```
ALTER TABLE qrtz_job_details ALTER COLUMN is_nonconcurrent SET NOT NULL;
ALTER TABLE qrtz_job_details ALTER COLUMN is_update_data SET NOT NULL;
```

**Comment by Eccenux** [comment]
I see no change.

**Comment by Eccenux** [comment]
Hey @stale bot that was rude 🙃. I said it was not fixed.
